### DISCUSSION TITLE
How to extend modeler with custom features

### DISCUSSION THREAD

**Discussion by mahdiSaeediN** [imixs/imixs-workflow#814] [Q&A]
Hello,

I'm looking for a way to provide BPMN model designer in Eclipse with additional functionality based on the existing model elements on canvas, so that when the user selects an element, I can access the node object clicked on, and be able to get the list of connected nodes to the label and so on. As an extension to this, I want to also customize context menu when right-clicking on elements, so more options appear on the list and assign functionality to those options. Are these possible with the imixs plug-in? and how is this different than the extension possibility of the BPMN2 modeler plug-in iteself?

Thanks in advance.

**Comment by rsoika** [comment]
Hi @mahdiSaeediN 
as mentioned in short [here](https://www.eclipse.org/forums/index.php/t/1112837/) I think the best way is to take a closer look into the new [Open-BPMN project](https://www.open-bpmn.org/). This modelling platform allows you to customize the modelling in various ways.

The one idea is to use the [Extension mechansim](https://www.open-bpmn.org/glsp-server/BPMN_EXTENSIONS.html). This allows you to extend your model objects with additional properties as covered by the BPMN 2.0 standard.

The other way is to extend the UI itself via the GLSP technology. This means you can add UI extensions which are reacting on modelling events like selecting an element or moving an element or creating a new connection. You also can add new visual elements like we did this with the Property panel. 

Let me know if this answers your question.
